### PR TITLE
Refactor SHA block to be prenorm residual, with layerscale at end of each branch

### PR DESCRIPTION
**Pull request by lucidrains**
Seems to be stable, even without differential learning rates

### CODE REVIEW

**Comment by lucidrains** [comment]
should close the gap between baseline in earlier epochs

File: bonito/models/configs/dna_r9.4.1@v3.1.toml [hunk 1]
```diff
@@ -21,6 +21,8 @@ features = 768
 rnn_type = "lstm"
 activation = "swish"
 blank_score = 2.0
+single_head_layers = [ 2, 3 ]
+single_head_gradient_mult = 1.0
```
Comment (lucidrains) [inline bonito/models/configs/dna_r9.4.1@v3.1.toml:25]: this shouldn't be needed, but just in case you run into stability, lower this to 0.2

File: bonito/nn.py [hunk 2]
```diff
@@ -127,6 +128,144 @@ def to_dict(self, include_weights=False):
...
+@register
+class FeedForward(Module):
+
+    def __init__(self, dim, mult=4, dropout=0.):
+        super().__init__()
+        self.net = nn.Sequential(
+            nn.LayerNorm(dim),
+            nn.Linear(dim, dim * mult),
+            nn.GELU(),
+            nn.Dropout(droput),
```
Comment (iiSeymour) [inline bonito/nn.py:212]: Typo -> `s/droput/dropout/`
